docs/do-not-merge.md

```diff
@@ -0,0 +1,815 @@
+|`compact/segmentAnalyzer/fetchAndProcessMillis`|Time taken to fetch and process segments to infer the schema for the compaction task to run.|`dataSource`, `taskId`, `taskType`, `groupId`,`tags`| Varies. A high value indicates compaction tasks will speed up from explicitly setting the data schema. |
+
+[#14752](https://github.com/apache/druid/pull/14752)
+
+### Segment scan metrics
+
+Added a new metric to figure out the usage of `druid.processing.numThreads` on the Historicals/Indexers/Peons.
+
+|Metric|Description|Dimensions|Normal value|
+|------|-----------|----------|------------|
+|`segment/scan/active`|Number of segments currently scanned. This metric also indicates how many threads from `druid.processing.numThreads` are currently being used.||Close to `druid.processing.numThreads`|
+
+### New Kafka consumer metrics
+
+Added the following Kafka consumer metrics:
+
+* `kafka/consumer/bytesConsumed`: Equivalent to the Kafka consumer metric `bytes-consumed-total`. Only emitted for Kafka tasks.
+* `kafka/consumer/recordsConsumed`: Equivalent to the Kafka consumer metric `records-consumed-total`. Only emitted for Kafka tasks.
+
+[#14582](https://github.com/apache/druid/pull/14582)
+
+### service/heartbeat metrics
+
+* Exposed `service/heartbeat` metric to `statsd-reporter` [#14564](https://github.com/apache/druid/pull/14564)
+* Modified the `service/heartbeat` metric to expose the `leader` dimension [#14593](https://github.com/apache/druid/pull/14593)
+
+### Tombstone and segment counts
+
+Added `ingest/tombstones/count` and `ingest/segments/count` metrics in MSQ to report the number of tombstones and segments after Druid finishes publishing segments.
+
+[#14980](https://github.com/apache/druid/pull/14980)
+
+## Extensions
+
+### Ingestion task payloads for Kubernetes
+
+You can now provide compressed task payloads larger than 128 KB when you run MiddleManager-less ingestion jobs.
+
+[#14887](https://github.com/apache/druid/pull/14887)
+
+### Prometheus emitter
+
+The Prometheus emitter now supports a new optional configuration parameter, `druid.emitter.prometheus.extraLabels`.
+This addition offers the flexibility to add arbitrary extra labels to Prometheus metrics, providing more granular control in managing and identifying data across multiple Druid clusters or other dimensions.
+For more information, see [Prometheus emitter extension](https://druid.apache.org/docs/latest/development/extensions-contrib/prometheus).
+
+[#14728](https://github.com/apache/druid/pull/14728)
+
+## Documentation improvements
+
+We've moved Jupyter notebooks that guide you through query, ingestion, and data management with Apache Druid to the new [Learn Druid](https://github.com/implydata/learn-druid) repository.
+The repository also contains a Docker Compose file to get you up and running with a learning lab.
+
+[#15136](https://github.com/apache/druid/pull/15136)
+
+# Upgrade notes and incompatible changes
+
+## Upgrade notes
```

Comment:
I think we should do something a bit different than previous releases before we finally publish these release notes, where this section is about upgrading from Druid 27, and then we also should include the upgrade notices from the previous few releases as well, just copied in from earlier release notes. So that way we have all the information in one place to cover like a year worth of upgrades. I don't think this needs to be added yet, we can wait until the last minute before publishing, but I think it would be helpful since operators don't always upgrade every release